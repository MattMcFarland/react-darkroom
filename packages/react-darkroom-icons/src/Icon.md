Icon elements are set by the `kind` prop
    <section>
      <Icon kind="floppyDisk"/>
      <Icon kind="undo"/>
      <Icon kind="undo2"/>
      <Icon kind="redo"/>
      <Icon kind="redo2"/>
      <Icon kind="crop"/>
      <Icon kind="image"/>
      <Icon kind="cross"/>
      <Icon kind="checkmark"/>
    </section>