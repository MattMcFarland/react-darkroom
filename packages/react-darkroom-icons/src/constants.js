export const iconKinds = {
  FLOPPYDISK: 'floppyDisk',
  UNDO: 'undo',
  UNDO2: 'undo2',
  REDO: 'redo',
  REDO2: 'redo2',
  CROP: 'crop',
  IMAGE: 'image',
  CROSS: 'cross',
  CHECKMARK: 'checkmark'
}
