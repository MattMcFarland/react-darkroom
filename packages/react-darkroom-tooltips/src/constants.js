export default ({
  NW: 'nw',
  N: 'n',
  NE: 'ne',
  E: 'e',
  SE: 'se',
  S: 's',
  SW: 'sw',
  W: 'w',
});
