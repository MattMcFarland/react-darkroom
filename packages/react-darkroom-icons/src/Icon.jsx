import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setDisplayName } from 'recompose';
import injectSheet from 'react-jss';
import iconStyles from './icon-styles';
import { iconKinds } from './constants';

const Icon = ({
  children,
  classes,
  kind,
}) => (<span className={classes[kind]}>{children}</span>);

Icon.propTypes = {
  /** One of available icon types (see example) */
  kind: PropTypes.oneOf(Object.values(iconKinds)).isRequired,
  /** react-jss classes */
  classes: PropTypes.shape(iconStyles).isRequired,
  children: PropTypes.node.isRequired,
};

export default compose(
  pure,
  injectSheet(iconStyles),
  setDisplayName('Icon'),
)(Icon);

