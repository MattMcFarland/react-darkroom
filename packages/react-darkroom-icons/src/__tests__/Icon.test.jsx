import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';
import constants from '../constants';

const { describe, test, expect } = global;

const createIconTests = kinds =>
  Object.entries(kinds).map(([key, value]) =>
    ({ constantKey: key, mapsTo: value, node: <Icon key={key} kind={value} /> }));

describe('<Icon />', () => {
  const iconTests = createIconTests(constants);
  describe('constants are used to map to classnames', () => {
    iconTests.forEach(({ constantKey, mapsTo, node }) => {
      test(`${constantKey} maps to kind: ${mapsTo}`, () => {
        expect(renderer.create(node)).toMatchSnapshot();
      });
    });
  });
});

