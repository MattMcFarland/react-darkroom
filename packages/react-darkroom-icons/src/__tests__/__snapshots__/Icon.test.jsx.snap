// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> constants are used to map to classnames CHECKMARK maps to kind: checkmark 1`] = `
<span
  className="checkmark-0-8"
/>
`;

exports[`<Icon /> constants are used to map to classnames CROP maps to kind: crop 1`] = `
<span
  className="crop-0-5"
/>
`;

exports[`<Icon /> constants are used to map to classnames CROSS maps to kind: cross 1`] = `
<span
  className="cross-0-7"
/>
`;

exports[`<Icon /> constants are used to map to classnames FLOPPYDISK maps to kind: floppyDisk 1`] = `
<span
  className="floppyDisk-0-0"
/>
`;

exports[`<Icon /> constants are used to map to classnames IMAGE maps to kind: image 1`] = `
<span
  className="image-0-6"
/>
`;

exports[`<Icon /> constants are used to map to classnames REDO maps to kind: redo 1`] = `
<span
  className="redo-0-3"
/>
`;

exports[`<Icon /> constants are used to map to classnames REDO2 maps to kind: redo2 1`] = `
<span
  className="redo2-0-4"
/>
`;

exports[`<Icon /> constants are used to map to classnames UNDO maps to kind: undo 1`] = `
<span
  className="undo-0-1"
/>
`;

exports[`<Icon /> constants are used to map to classnames UNDO2 maps to kind: undo2 1`] = `
<span
  className="undo2-0-2"
/>
`;
